mavros/src/lib/mavros.cpp

```diff
@@ -135,8 +139,14 @@ MavRos::MavRos() :
 		mavlink_pub_cb(msg, framing);
 		plugin_route_cb(msg, framing);
 
-		if (gcs_link)
+		if (gcs_link) {
+			if (this->gcs_quiet_mode && msg->msgid != mavlink::common::msg::HEARTBEAT::MSG_ID &&
+				(ros::Time::now() - this->last_message_received_from_gcs > ros::Duration(20))) {
```

Comment:
The duration should be a parameter aswell.

Reply:
I'm not sure this should be a parameter, because it's too many parameters. @voon?

Reply:
If you are adding new features and adding new params for them, at least make the implementation flexible so the user can setup the feature itself without having to rebuild the code. Please have this changed.